a10networks/a10-octavia
# [DB-Migration][Stack-1718] Added a10-neutron-lbaas scripts to our db

## Description
- For smooth db migration from a10-neutron-lbaas to a10-octavia, to maintain backward compatibility as well as avoiding manual migrations.

## Jira Ticket
https://a10networks.atlassian.net/browse/STACK-1718

## Technical Approach
- Copied all db scripts from a10-neutron-lbaas to a10-octavia db

## Config Changes
None

## Test Cases
- This is an incomplete feature of a large migration, hence only test case is, while running alembic upgrade head, it should generate all a10-neutron-lbaas tables for us in a pointed db.



## Manual Testing
1. Create a db in mysql similarly as above steps. eg - a10_octavia_dummy.

2. changed env.py file to create new  alembic_version_a10 table inside a10_octavia_dummy db.

3. modify `/etc/octavia/octavia.conf`  `[database]` section config and set `connection`  to point to `a10_octavia_dummy` db.

4. run `alembic upgrade head`.
You will see all a10-neutron-lbaas tables has been created inside dummy db along with alembic_version_a10 table.

![image](https://user-images.githubusercontent.com/52992745/97949873-01ba2b80-1dbb-11eb-94df-bcc3d6455919.png)
